ITSNP-4: Design profile components (m4)

ace4port opened the pull request:
Avatar and Profile components added

- added header
- added foot
- added profile Card
- shame: Profile comments icon need to be optimized/ changed
- added Avatar component
- shame: Avatar set conditional rendering in scss not js
- shame: Avatar size variants medium and large from scss

16 June Update
Profile componenets 1,2,3
![image](https://user-images.githubusercontent.com/39233007/122250428-60b35780-cee9-11eb-8aa3-c04f1b88bddd.png)
Comment components
![image](https://user-images.githubusercontent.com/39233007/122250584-804a8000-cee9-11eb-8f93-10a8b1d2b23c.png)

ace4port committed:
ITSON PROF-001 Created profile components AV-001 Created avatar components
-added header
added foot
-added profile Card
-added Avatar component
shame: Avatar set conditional rendering in scss not js
shame: Avatar size variants medium and large from scss

SurrealSerenade commented:
@ace4port Can you please link the issues related to this pull request?

SurrealSerenade commented on src/components/Avatar/Avatar.js:
```diff
@@ -0,0 +1,18 @@
+import defImage from "../../Assets/avatarDefault.png";
+import "./Avatar.scss";
+
+//this component expects a link to image source and not passed displays a default image
```
Please, use multi-line commenting when documenting components. You can even specify the author if you want to.

SurrealSerenade commented on src/components/Avatar/Avatar.scss:
```diff
@@ -0,0 +1,20 @@
+}
+.dotted {
+  img {
+    border: 2px dashed black; // add logic to make this line conditional
```
Problem: We can't use conditional statements in CSS.
Suggestion: We can use modifiers (check out BEM), then implement a conditional statement on JS to specify modifiers for this particular problem.

SurrealSerenade commented on src/components/Avatar/Avatar.scss:
```diff
@@ -0,0 +1,20 @@
+  img {
+    border-radius: 50%;
+    width: 50px;
+    height: 50px; //70px
```
Can you explain why height is specified as 50px but also commented as 70px? Thank you.

RudeSoul replied:
@ace4port its better to use REM rather than PX,

ace4port replied:
In templates, there are 4 varying sizes for avatar- 2 each are of similar size (70 67, 48 50)
So I thought to make it conditional or use modifiers to display 2 variants. Comment serves for reminder for me

ace4port replied:
Done

SurrealSerenade commented on src/Assets/Icons/Icons.js:
```diff
@@ -0,0 +1,33 @@
+import React from "react";
+
+export const HeartIcon = () => {
```
@RudeSoul @ace4port  Are we allowing the implementation of multiple components in one file? I think it should be one component per file, make a directory for all relative files and then export them using an index.js file.
Doing so will help in the following ways:
- Keeps the code modular which can be used in other projects if needed
- Searching for a specific component in the module will be easier
- Keeps the code clean and easier to review each component

An example of this can be seen in [Material UI](https://github.com/mui-org/material-ui/tree/next/packages/material-ui-icons/lib)

RudeSoul replied:
I don't think we are allowing the implementation fo multiple components in one file. 
@ace4port  Please resolve it

SurrealSerenade commented on src/components/Profile/Profile.js:
```diff
@@ -0,0 +1,62 @@
+
+import { HeartIcon, CommentIcon, FavouriteIcon } from "../../Assets/Icons/Icons";
+
+export const ProfileTop = ({ image, authorName, followerNum }) => {
```
@RudeSoul @ace4port Similar to Icons.js

RudeSoul replied:
`ProfileTop` Naming could be better

ace4port replied:
Done

SurrealSerenade commented on src/components/Profile/Profile.scss:
```diff
@@ -0,0 +1,73 @@
+}
+
+@mixin button {
+  // background-color: #000;
```
Please, remove the CSS if they are not required.

SurrealSerenade commented on src/pages/Home/Home.js:
```diff
@@ -1,14 +1,16 @@
+import { ProfileTop, ProfileBottom, ProfileCard } from "../../components/Profile/Profile";
 
-import './Home.scss';
+import avatarImg from "../../Assets/6hk992di5pn61.jpg";
```
I can see that you have used image: 6hk992di5pn61.jpg for preview. You can use placeholder image links from [Picsum](https://picsum.photos/) or [Placeholder](https://placeholder.com/) or any other image placeholder sites.
Remove the asset (from repo) as this was not required and replace it with any placeholder image link for preview.

SurrealSerenade reviewed (changes requested):
Please, have a look at the comments in PR. Thank you.

RudeSoul commented:
Have a look over it, There are the most important things to be noted, Please resolve them so that we could review them.

![image](https://user-images.githubusercontent.com/30189506/121337782-9c27b200-c93c-11eb-836f-16b019319f62.png)

RudeSoul commented:
Please look at the naming convention of PR TITLE, It's inappropriate. 

![image](https://user-images.githubusercontent.com/30189506/121338219-03ddfd00-c93d-11eb-8442-ea924ba4d642.png)

RudeSoul commented:
If you have shame work, Keep the pr in WIP || Draft

![image](https://user-images.githubusercontent.com/30189506/121338752-86ff5300-c93d-11eb-8dcb-1d1bf10b3734.png)

RudeSoul commented on src/components/Profile/Profile.js:
```diff
@@ -0,0 +1,62 @@
+
+export const ProfileTop = ({ image, authorName, followerNum }) => {
+  return (
+    <div className='profile profileTopWrap'>
```
Where is BEM?

RudeSoul commented on src/components/Profile/Profile.js:
```diff
@@ -0,0 +1,62 @@
+  return (
+    <div className='profile profileTopWrap'>
+      <Avatar image={image} dotted={true} />
+      <h4 className='title'>{authorName || "John Doe"}</h4>
```
Where is BEM?  className could be `profile__title`

RudeSoul commented on src/components/Profile/Profile.js:
```diff
@@ -0,0 +1,62 @@
+  return (
+    <div className='profile profileBottomWrap'>
+      <Avatar image={image} />
+      <div className='authorInfo'>
```
same as above, where is BEM?

RudeSoul commented on src/components/Profile/Profile.js:
```diff
@@ -0,0 +1,62 @@
+      <Avatar image={image} />
+      <div className='authorInfo'>
+        <p>
+          <em>Author</em>
```
why `<em>`? This is why you should link the issue. We do not know the design of this part, so it's getting tedious to review. Please fix

RudeSoul commented on src/components/Profile/Profile.js:
```diff
@@ -0,0 +1,62 @@
+        <p>
+          <em>Author</em>
+        </p>
+        <h4 className='title'>{authorName || "John Doe"}</h4>
```
Where is BEM?

RudeSoul commented on src/components/Profile/Profile.js:
```diff
@@ -0,0 +1,62 @@
+        <h4 className='title'>{authorName}</h4>
+        <p className='desc'>
+          {authorDesc ||
+            "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."}
```
As far as possible, do not use Lorem. Even the computer does not understand lorem.

RudeSoul commented on src/components/Profile/Profile.js:
```diff
@@ -0,0 +1,62 @@
+          {authorDesc ||
+            "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."}
+        </p>
+        {/* <br /> */}
```
Remove unwanted comments or have some info message on it, why it is there even after been commented.

RudeSoul commented on src/components/Profile/Profile.js:
```diff
@@ -0,0 +1,62 @@
+      </div>
+    </>
+  );
+};
```
You have three components on a single page, ### Why?

RudeSoul commented on src/components/Profile/Profile.js:
```diff
@@ -0,0 +1,62 @@
+export const ProfileCard = ({ image, authorName, authorDesc, followerNum, commentsNum }) => {
+  return (
+    <>
+      <div className='profile card'>
```
why do we have classNames like `profile` `card` ?? These could be more descriptive.

RudeSoul commented on src/components/Profile/Profile.scss:
```diff
@@ -0,0 +1,73 @@
+@import "~styles/main";
+
+@mixin title {
```
why do we have mixin' over here?
Mixins are global and used wherever needed, that's is why we do have separate `_mixins.scss` file for mixins

ace4port replied:
I dont have much experience with css. And this is first time using sass. So apologies

RudeSoul commented on src/components/Profile/Profile.scss:
```diff
@@ -0,0 +1,73 @@
+  font-size: 18px;
+}
+
+@mixin button {
```
same as above, It's critical. Please fix

RudeSoul commented on src/components/Profile/Profile.scss:
```diff
@@ -0,0 +1,73 @@
+  }
+}
+
+.profileTopWrap {
```
it's not acceptable, You have to follow BEM strictly.

RudeSoul commented on src/components/Profile/Profile.scss:
```diff
@@ -0,0 +1,73 @@
+  align-items: row;
+}
+
+.profileBottomWrap {
```
it's not acceptable, You have to follow BEM strictly.

RudeSoul commented on src/components/Profile/Profile.scss:
```diff
@@ -0,0 +1,73 @@
+}
+
+.card {
+  width: 210px;
```
It's not good to use fixed-width, If you had linked the issue I would suggest some solution, but for now sorry I do not have a solution, and fix width is not accept most of the time.
Please fix

RudeSoul commented on src/components/Profile/Profile.scss:
```diff
@@ -0,0 +1,73 @@
+  .desc {
+    color: color(light);
+  }
+  .iconContainer {
```
same as above

RudeSoul commented on src/components/Profile/Profile.js:
```diff
@@ -0,0 +1,62 @@
+    <div className='profile profileTopWrap'>
+      <Avatar image={image} dotted={true} />
+      <h4 className='title'>{authorName || "John Doe"}</h4>
+      <Button label={`Follow ${followerNum || "1.2"}k`} />
```
even the `k` must come with a prop, Think like what if there are only 2 followers. For example: With your code, the person having 2 followers will have 2K followers printed. 
And we don't do fake data

RudeSoul commented on src/components/Profile/Profile.scss:
```diff
@@ -0,0 +1,73 @@
+
+@mixin title {
+  margin: auto 20px auto 0px;
+  color: #000;
```
we do not use color like this.

SurrealSerenade commented on src/Assets/Icons/Icons.js:
```diff
@@ -0,0 +1,33 @@
+import React from "react";
```
Can you explain why Icons is in the Assets folder?

RudeSoul commented on src/components/Profile/Profile.scss:
```diff
@@ -0,0 +1,73 @@
+    display: flex;
+    justify-content: space-between;
+    margin: 2px 4px;
+    color: color(light);
```
The naming of color is not appropriate, color name `light` is not descriptive.

SurrealSerenade replied:
Also, this color does not exist.

ace4port replied:
Done

RudeSoul commented:
Explain your commit a little bit more rather than just in two words,
And please do commit in list like `https://github.com/itsnporg/blog-itsnp-frontend/pull/11#issue-663762198` here

![image](https://user-images.githubusercontent.com/30189506/121344912-fa0bc800-c943-11eb-9b98-4f3f507c7faa.png)

SurrealSerenade commented:
> Please look at the naming convention of PR TITLE, It's inappropriate.
> 
> ![image](https://user-images.githubusercontent.com/30189506/121338219-03ddfd00-c93d-11eb-8442-ea924ba4d642.png)

You can use ITSNP-ISSUE_NUMBER: ISSUE_TITLE
example: ITSNP-1: Design Nav-bar

ace4port committed:
ITSNP-4: Profile Components
-resolved multiple exports from one components
-removed assets folder
-made mixins for avatar and made it render conditionally
-resolved shames

ace4port committed:
ITSON-4: Profile/ Comment compenents
-WIP comment componet styles progress
fixed react error in icon styles

RudeSoul commented on src/components/Avatar/Avatar.js:
```diff
@@ -0,0 +1,20 @@
+let defaultPlaceholder = "https://i.picsum.photos/id/1081/200/300.jpg?hmac=ntCnXquH7cpEF0vi5yvz1wKAlRyd2EZwZJQbgtfknu8";
+
+//this component expects avatar image from backend
+// if not passed, it displays a default image
```
this is not multi-line comments, [Visit this site for reference](https://wesbos.com/react-jsx-comments/)

ace4port committed:
ITSNP-4: Profile Components
- added comment component
- fixed some inconsistencies
- added proper documentation
- WIP: font is inconsistent
- SHAME: very hacky fixes used to align icons in comment buttom

SurrealSerenade commented on src/components/Avatar/Avatar.js:
```diff
@@ -0,0 +1,23 @@
+
+import "./Avatar.scss";
+
+let defaultPlaceholder = "https://i.picsum.photos/id/1081/200/300.jpg?hmac=ntCnXquH7cpEF0vi5yvz1wKAlRyd2EZwZJQbgtfknu8";
```
Put the placeholder inside the component, and also make it const.

SurrealSerenade commented on src/components/Avatar/Avatar.scss:
```diff
@@ -0,0 +1,17 @@
+@import "~styles/main";
+
+.dotted {
```
Haven't I shown you how to do this in call? You can also checkout the stackblitz [link](https://stackblitz.com/edit/react-cnpjqt).

ace4port replied:
Done

RudeSoul commented on src/components/Avatar/Avatar.js:
```diff
@@ -1,15 +1,20 @@
     <>
-      <div className={`${dotted ? "dotted" : "container"}`}>
-        <img src={image || defImage} alt='AvatarImage' />
+      <div className={`container ${dotted ? "dotted" : variant === "lg" ? "lg" : "sm"}`}>
```
container?

RudeSoul commented on src/components/Avatar/Avatar.scss:
```diff
@@ -1,20 +1,17 @@
 
-.container {
+.dotted {
   img {
```
does it adjust all sizes of images?

ace4port replied:
![image](https://user-images.githubusercontent.com/39233007/121918368-b48d3780-cd55-11eb-9a02-feb1b5b03bb2.png)
Made images look like this

SurrealSerenade commented on src/components/Comment/Comments.scss:
```diff
@@ -0,0 +1,74 @@
+    margin: 0px -60px;
+  }
+
+  &__top {
```
There already exists mixin to create the element. Why are you not using that mixin? Also, 'top' as a name does not signify its function, use a better name.

RudeSoul replied:
Use mixins, Visit `~/src/styles/utils/_mixins.scss`

RudeSoul commented on src/components/Comment/Comments.scss:
```diff
@@ -0,0 +1,74 @@
+  //shame: this is very hacky method. If found an alternative, please replace
+  .avatar {
+    position: fixed;
+    margin: 0px -60px;
```
This needs to be fixed, If you have this type of code and still want a reviewer for review, Keep the pr in WIP and request for review.

SurrealSerenade commented on src/components/Comment/Comments.scss:
```diff
@@ -0,0 +1,74 @@
+    padding: 0;
+    .comment__author {
+      font-weight: 700;
+      font-size: 0.75rem;
```
Why use different types of units? You have used pixels and rem both, but why?

RudeSoul replied:
either use px or rem, not both

SurrealSerenade commented on src/components/Comment/Comments.scss:
```diff
@@ -0,0 +1,74 @@
+    align-items: center;
+    justify-content: space-between;
+    font-size: 16px;
+    iconContainer {
```
Where is BEM?

SurrealSerenade commented on src/components/Comment/Comments.scss:
```diff
@@ -0,0 +1,74 @@
+    }
+    .author {
+      vertical-align: middle;
+      display: inline-block;
```
Using `vertical align:middle; display:inline-block` just to center is redundant.

RudeSoul commented on src/components/Comment/Comments.scss:
```diff
@@ -0,0 +1,74 @@
+      font-size: 0.75rem;
+      display: inline-block;
+      vertical-align: middle;
+      // margin-right: 200px;   // shame- cant give margin right of text
```
if shame is resolved, it's better to remove it rather than commenting it.
Or if you want it as a comment, please give a description so that someone from the future will not get confused

RudeSoul commented on src/components/Comment/Comments.scss:
```diff
@@ -0,0 +1,74 @@
+      vertical-align: middle;
+      // margin-right: 200px;   // shame- cant give margin right of text
+    }
+    em {
```
do not style element, we have `className` in react for it

RudeSoul commented on src/components/Comment/Comments.scss:
```diff
@@ -0,0 +1,74 @@
+      display: block;
+      color: lighten(color(pure-black), 50%);
+    }
+    &__detailsIcon {
```
visit `~/src/styles/utils/_mixins.scss` and use element mixin from there.

RudeSoul commented on src/components/Comment/Comments.scss:
```diff
@@ -0,0 +1,74 @@
+    em {
+      font-weight: 400;
+      font-size: 0.66rem;
+      display: block;
```
If em is block you does not need to define display block, if em is not block do not make non block element block

ace4port replied:
em is inline element. Not defining block makes is display on same line as author name.

RudeSoul replied:
than we have Flex for that, But making inline element as block is bad idea

RudeSoul commented on src/components/Comment/Comments.scss:
```diff
@@ -0,0 +1,74 @@
+      vertical-align: middle;
+      display: inline-block;
+    }
+    span {
```
do not style element, we have `className` in react for it

RudeSoul commented on src/components/Comment/Comments.scss:
```diff
@@ -0,0 +1,74 @@
+    }
+  }
+
+  &__mid {
```
visit ~/src/styles/utils/_mixins.scss and use element mixin from there.

RudeSoul commented on src/components/Comment/Comments.scss:
```diff
@@ -0,0 +1,74 @@
+    color: lighten(color(pure-black), 20%);
+  }
+
+  &__bot {
```
visit ~/src/styles/utils/_mixins.scss and use element mixin from there.

RudeSoul commented on src/components/Comment/Comments.scss:
```diff
@@ -0,0 +1,74 @@
+      display: inline-block;
+    }
+    span {
+      box-sizing: border-box;
```
why this?

SurrealSerenade commented on src/components/Comment/Comments.scss:
```diff
@@ -0,0 +1,74 @@
+      display: inline-block;
+      color: lighten(color(pure-black), 40%);
+    }
+    .reply {
```
Where is BEM?

RudeSoul commented on src/components/Comment/Comments.scss:
```diff
@@ -0,0 +1,74 @@
+    }
+    span {
+      box-sizing: border-box;
+      margin: 5px;
```
the `margin` doesn't function properly on the span, the only `margin-left`, and `margin-right` works

RudeSoul commented on src/components/Comment/Comments.scss:
```diff
@@ -0,0 +1,74 @@
+      margin: 5px;
+      line-height: 24px;
+    }
+    &__name {
```
visit ~/src/styles/utils/_mixins.scss and use element mixin from there.

SurrealSerenade commented on src/components/Comment/index.js:
```diff
@@ -0,0 +1,63 @@
+      <div className='comment__mid'>
+        <p>
+          {comment.contents ||
+            `Life is a beautiful journey that is meant to be embraced to the fullest every day. However, that doesn’t mean
```
Do we need placeholder comments when no comment exists?

RudeSoul commented on src/components/Comment/Comments.scss:
```diff
@@ -0,0 +1,74 @@
+      color: lighten(color(pure-black), 40%);
+    }
+    .reply {
+      font-size: 0.85rem;
```
make your units consistent, don't do haphazardly. either use `PX` or `REM` not both

ace4port replied:
Using rem or em for fonts is ok but not recommended for setting margins, padding
https://stackoverflow.com/questions/45822322/when-to-choose-rem-over-px#:~:text=You%20can%20use,are%20mostly%20used.
If there is specific reason to use rem for margin, I could change

RudeSoul replied:
You still need more research for a specific reason (Y)

[Read this - 1](https://graphicdesign.stackexchange.com/questions/2147/which-unit-should-be-used-in-web-design-to-define-space-margin-padding-etc)

[Read this - 2](https://daveredfern.com/use-units-css/)

[Read this - 3](https://gist.github.com/basham/2175a16ab7c60ce8e001)

[Read this - 4](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Values_and_units)

[Read this - 5](https://medium.com/swlh/css-units-which-ones-to-use-and-avoid-31e4ed461f9)

[Read this - 6](https://www.digitalocean.com/community/tutorials/how-to-use-common-units-in-css)

[Read this - 7](https://www.digitalocean.com/community/tutorials/css-css-units-explained)

[Read this - 8](https://labs.tadigital.com/index.php/2020/06/18/common-units-in-css-and-when-to-use/)

[Read this - 9](https://getflywheel.com/layout/choose-css-unit-create-better-site-layouts-how-to/)

[Read this - 10](https://daveredfern.com/use-units-css/)

[Read this - 11](https://css-tricks.com/confused-rem-em/)


Hope you get the answer, And please share with us, what should we need to use :)

RudeSoul replied:
You still need more research for a specific reason (Y)

[Read this - 1](https://graphicdesign.stackexchange.com/questions/2147/which-unit-should-be-used-in-web-design-to-define-space-margin-padding-etc)

[Read this - 2](https://daveredfern.com/use-units-css/)

[Read this - 3](https://gist.github.com/basham/2175a16ab7c60ce8e001)

[Read this - 4](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Values_and_units)

[Read this - 5](https://medium.com/swlh/css-units-which-ones-to-use-and-avoid-31e4ed461f9)

[Read this - 6](https://www.digitalocean.com/community/tutorials/how-to-use-common-units-in-css)

[Read this - 7](https://www.digitalocean.com/community/tutorials/css-css-units-explained)

[Read this - 8](https://labs.tadigital.com/index.php/2020/06/18/common-units-in-css-and-when-to-use/)

[Read this - 9](https://getflywheel.com/layout/choose-css-unit-create-better-site-layouts-how-to/)

[Read this - 10](https://daveredfern.com/use-units-css/)

[Read this - 11](https://css-tricks.com/confused-rem-em/)

and do check out what these popular frameworks use,

[bootstrap](https://getbootstrap.com/docs/5.0/utilities/spacing/)

[Foundation](https://get.foundation/sites/docs/global.html)

[TailwindCSS Margin](https://tailwindcss.com/docs/margin)

[TailwindCSS Padding](https://tailwindcss.com/docs/padding)

[Styled Component](https://styled-components.com/docs/advanced)


Hope you get the answer. And **please recommend**, what we should use :)

SurrealSerenade commented on src/components/Comment/index.js:
```diff
@@ -0,0 +1,63 @@
+            <span>{author.followerNum || 0}</span>
+            {/* Shame: Cant think of better name- comment list refers to people who have commented */}
+            <div className='comment__bot__name'>
+              {comment.list || "Subedi bibek, BOT, Awebisam"} and {205} others
```
When we cannot find people who like the comment, do we tell them 'Subedi bibek, BOT, Awebisam and 205 others like your post? If so, why? Also, can't className be comment__likeList or comment__likes?

RudeSoul commented on src/components/Comment/index.js:
```diff
@@ -0,0 +1,63 @@
+          <div className='author'>
+            <span>{author.followerNum || 0}</span>
+            {/* Shame: Cant think of better name- comment list refers to people who have commented */}
+            <div className='comment__bot__name'>
```
BEM stands for `Block__Element–Modifier`, not `Block__Element__Element–Modifier`. So, avoid multiple-element level naming. If you’re getting to great-great-great-grandchild levels, then you’ll probably want to revisit your component structure anyway.

RudeSoul commented on src/components/Icons/ThreeDots.js:
```diff
@@ -0,0 +1,24 @@
+ *
+ * @returns 3 dots/ details icon component
+ */
+//shame: Naming could be better? Please suggest
```
might be moreIcon

SurrealSerenade commented on src/components/Icons/icons.scss:
```diff
@@ -0,0 +1,17 @@
+@import "~styles/main";
+
+.icon {
+  display: inline-block;
```
Why do we even need this scss?

SurrealSerenade commented on src/components/Icons/icons.scss:
```diff
@@ -0,0 +1,17 @@
+}
+//shame: this is very hacky method. If found an alternative, please replace
+.icons {
+  display: inline-block;
```
Why do we even need this code?

RudeSoul commented on src/components/Icons/icons.scss:
```diff
@@ -0,0 +1,17 @@
+//shame: this is very hacky method. If found an alternative, please replace
+.icons {
+  display: inline-block;
+  margin: 0 0px;
```
it could be `margin:0`

SurrealSerenade commented on src/components/Profile/AuthorProfileBottom.js:
```diff
@@ -0,0 +1,22 @@
+ */
+export const AuthorProfileBottom = ({ image, authorName }) => {
+  return (
+    <div className='profile profileBottom'>
```
Why not use BEM?

SurrealSerenade commented on src/components/Profile/AuthorProfileBottom.js:
```diff
@@ -0,0 +1,22 @@
+      <Avatar image={image} variant='lg' dotted={true} />
+      <div className='authorInfo'>
+        <p>Author</p>
+        <h4 className='profile__title'>{authorName || "John Doe"}</h4>
```
When we can't find author's name, are we gonna show 'John Doe' as the author?

RudeSoul replied:
very bad BEM

SurrealSerenade commented on src/components/Profile/AuthorProfileCard.js:
```diff
@@ -0,0 +1,39 @@
+          <Button label='Follow' />
+        </div>
+        <h4 className='profile__title'>{authorName}</h4>
+        <p className='desc'>
```
'desc' = Descending?

SurrealSerenade commented on src/components/Profile/AuthorProfileCard.js:
```diff
@@ -0,0 +1,39 @@
+        </div>
+        <h4 className='profile__title'>{authorName}</h4>
+        <p className='desc'>
+          {authorDesc || "This is some placeholder text. Kitten Cat Kitty Cat Kitty Cat Kitty Cat"}
```
So we gonna show 'This is some placeholder text. Kitten Cat Kitty Cat Kitty Cat Kitty Cat' as author Descending?

SurrealSerenade commented on src/components/Profile/AuthorProfileCard.js:
```diff
@@ -0,0 +1,39 @@
+            <HeartIcon />
+            <span>{followerNum}k</span>
+          </div>
+          <div className='heart'>
```
Does this contain a heart?

RudeSoul commented on src/components/Profile/AuthorProfileBottom.js:
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+
+import "./Profile.scss";
+import Avatar from "../Avatar/Avatar";
```
leave a space, try to make a group

RudeSoul commented on src/components/Profile/AuthorProfileCard.js:
```diff
@@ -0,0 +1,39 @@
+import React from "react";
+
+import "./Profile.scss";
+import Avatar from "../Avatar/Avatar";
```
leave a space, try to make a group

RudeSoul commented on src/components/Profile/AuthorProfileCard.js:
```diff
@@ -0,0 +1,39 @@
+  return (
+    <>
+      <div className='profile card'>
+        <div className='first'>
```
confusing class name

RudeSoul commented on src/components/Profile/AuthorProfileTop.js:
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+
+import "./Profile.scss";
+import Avatar from "../Avatar/Avatar";
```
leave a space, try to make a group

SurrealSerenade commented on src/components/Profile/Profile.scss:
```diff
@@ -0,0 +1,56 @@
+  margin: 10px 20px;
+  width: 100%;
+  display: flex;
+  align-items: row;
```
Is there a `row` value to align-items property?

ace4port replied:
It was a mistake. I have resolved

SurrealSerenade commented on src/components/Profile/Profile.scss:
```diff
@@ -0,0 +1,56 @@
+  .button {
+    margin: auto 2px;
+  }
+  .profile__title {
```
Use the available mixin.

ace4port replied:
Done

SurrealSerenade commented on src/styles/utils/_mixins.scss:
```diff
@@ -37,3 +36,14 @@
   }
 }
+
+@mixin avatar($size, $border) {
```
This mixin will not be allowed.

ace4port replied:
Removed

RudeSoul commented on src/components/Profile/Profile.scss:
```diff
@@ -1,59 +1,42 @@
   }
-  .title {
-    @include title;
+  .profile__title {
```
visit `~/src/styles/utils/_mixins.scss` and use element mixin from there

RudeSoul commented on src/components/Profile/Profile.scss:
```diff
@@ -1,59 +1,42 @@
+    margin: auto 20px auto 0px;
+    color: color(pure-black);
+    text-transform: capitalize;
+    font-size: 1.125rem;
```
again `rem` here

RudeSoul commented on src/components/Profile/Profile.scss:
```diff
@@ -1,59 +1,42 @@
 .card {
-  width: 210px;
+  width: 100%;
   .first {
```
visit `~/src/styles/utils/_mixins.scss` and use element mixin from there

RudeSoul commented on src/styles/utils/_mixins.scss:
```diff
@@ -12,10 +12,9 @@
 
-@mixin font($font-family, $font-weight, $font-style) {
-  font-family: $font-family;
+@mixin font($font-family: "Inter", $font-style: normal, $font-size: 24px, $font-weight: normal) {
```
Is `inter` the default font for our project?

RudeSoul commented on src/styles/utils/_variables.scss:
```diff
@@ -4,6 +4,7 @@ $colors: (
   tertiary-grey: #f8f8f8,
   primary-white: #ffffff,
   pure-black: #000000,
+  text-light: #00000099,
```
6 digits are limit, if you are trying to include opacity too Please don't

SurrealSerenade commented on src/styles/utils/_variables.scss:
```diff
@@ -4,6 +4,7 @@ $colors: (
   tertiary-grey: #f8f8f8,
   primary-white: #ffffff,
   pure-black: #000000,
+  text-light: #00000099,
```
Why do we have this color? The length of `text-light` color in hex code is 8. Which means last two denote opacity of 99 (60%). If this is required case use more descriptive color RBGA color model.

ace4port replied:
Removed

ace4port committed:
ITSNP-4: Design Profile components
- refactored components to make consistency
- removed avatar mixin
- changed default font to poppins
- moved defualt props to function declaration for cleaner logic
- some fixes

ace4port committed:
ITSNP-04: Design profile components
- refactored avatar class components
- added some bem
- renamed components to better names
- wip: Font, Margin units are inconsistent
- shame: Rem and px units used- need to refactor

ace4port committed:
ITSNP-04: Design Profile components
- change avatar component to make it follow bem convention

ace4port committed:
PROF-04: Design profile components
- used mixins following bem
- introduced new size xs for avatar used in other component
- added use in documentation comment
- organized border and size as saperate avatar modifiers

ace4port committed:
PROF-04: Design profile components
- removed lighten function

ace4port committed:
ITSNP04 Design profile components
- made avatar component render border and size on div instead of image
